### PR TITLE
Add a test for a parent selector in no-duplicate-selectors

### PR DESCRIPTION
**Pull request by hudochenkov**
> Which issue, if any, is this issue related to?

https://github.com/stylelint/stylelint/issues/2198

> Is there anything in the PR that needs further explanation?

No, it's self explanatory.

### CODE REVIEW

File: lib/rules/no-duplicate-selectors/__tests__/index.js [hunk 1]
```diff
@@ -84,6 +84,12 @@ testRule(rule, {
     message: messages.rejected("a b"),
     line: 1,
     column: 12,
+  }, {
+    code: "a { & {} }",
+    description: "duplicate caused by &-parent selector",
+    message: messages.rejected("a"),
```
Comment (davidtheclark) [inline lib/rules/no-duplicate-selectors/__tests__/index.js]: Wait ... why is this a duplicate? Don't we end up with `a` and `a a`, which are different selectors?

Comment (hudochenkov) [inline lib/rules/no-duplicate-selectors/__tests__/index.js]: It will be `a {} a {}` or `a {}`. We can't predict what preprocessor will do. Example http://www.sassmeister.com/gist/8ad7e0270cf5817804625228e74f9c6b

Comment (jeddy3) [inline lib/rules/no-duplicate-selectors/__tests__/index.js]: In the [CSS nesting instance](https://tabatkins.github.io/specs/css-nesting/#nest-selector) it looks like `&` is syntactic sugar for `:matches()`. So, I guess 

```
a { & {} }
```

Just becomes:

```
a {}
:matches(a) {}
```

Technically different selectors, but they do the same thing.

Anyhow, I think you're [excellent write-up of the edge-cases and inconsistencies](https://github.com/stylelint/stylelint/issues/2198#issuecomment-274276106) now makes me think we should just give any rule with just `&` as its selector a wide-berth i.e. don't resolve it and don't check it. If we go down this path, should we do it in stylelint or within `postcss-resolve-nested-selector`?

Comment (hudochenkov) [inline lib/rules/no-duplicate-selectors/__tests__/index.js]: Worth noting that [CSS Nesting Module Level 3](https://tabatkins.github.io/specs/css-nesting/) it's just a Tab Atkin's idea. It's not related to W3C. Basically it have the same “power” as any blog post.

I think postcss-resolve-nested-selector is working well and resolves `&` well.

Comment (davidtheclark) [inline lib/rules/no-duplicate-selectors/__tests__/index.js]: > we should just give any rule with just & as its selector a wide-berth i.e. don't resolve it and don't check it. If we go down this path, should we do it in stylelint or within postcss-resolve-nested-selector?

I think that for all intents and purposes `a { & { color: pink; } }` includes a duplicate selector, and over in #2198 everybody seems to agree about that. So it should be flagged.

However, because of #2198 and #2199 it looks like some people want an option to *not* consider the lone `&` to be a duplicate selector. That could be a new feature request.

**Comment by davidtheclark** [comment]
👍 merging!
